hbase-server/src/main/java/org/apache/hadoop/hbase/client/ClientSideRegionScanner.java

```diff
@@ -93,9 +93,12 @@ public ClientSideRegionScanner(Configuration conf, FileSystem fs,
   @Override
   public Result next() throws IOException {
     values.clear();
-    scanner.nextRaw(values);
-    if (values.isEmpty()) {
-      //we are done
+    boolean moreValues;
+    do {
+      moreValues = scanner.nextRaw(values);
+    } while (values.isEmpty() && moreValues);
+
+    if (!moreValues) {
```

Comment:
We will see empty result (values) but still more rows when the scan  met RPC timeout value?  Will it be possible to have  UT here?

Reply:
Scan is not an RPC timeout situation. After scannerContext.returnImmediately() is called in storeScanner, it is returned as true by scannerContext.checkTimeLimit(limitScope) in RegionScannerImpl.

https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RegionScannerImpl.java#L563